hudi-common/src/main/java/org/apache/hudi/internal/schema/utils/SchemaChangeUtils.java

```diff
@@ -72,38 +72,59 @@ private static boolean isTypeUpdateAllowInternal(Type src, Type dsr) {
     switch (src.typeId()) {
       case INT:
         return dsr == Types.LongType.get() || dsr == Types.FloatType.get()
-            || dsr == Types.DoubleType.get() || dsr == Types.StringType.get() || dsr.typeId() == Type.TypeID.DECIMAL;
+            || dsr == Types.DoubleType.get() || dsr == Types.StringType.get() || dsr.typeId() == Type.TypeID.DECIMAL || dsr.typeId() == Type.TypeID.DECIMAL_FIXED;
       case LONG:
-        return dsr == Types.FloatType.get() || dsr == Types.DoubleType.get() || dsr == Types.StringType.get() || dsr.typeId() == Type.TypeID.DECIMAL;
+        return dsr == Types.FloatType.get() || dsr == Types.DoubleType.get() || dsr == Types.StringType.get() || dsr.typeId() == Type.TypeID.DECIMAL || dsr.typeId() == Type.TypeID.DECIMAL_FIXED;
       case FLOAT:
-        return dsr == Types.DoubleType.get() || dsr == Types.StringType.get() || dsr.typeId() == Type.TypeID.DECIMAL;
+        return dsr == Types.DoubleType.get() || dsr == Types.StringType.get() || dsr.typeId() == Type.TypeID.DECIMAL || dsr.typeId() == Type.TypeID.DECIMAL_FIXED;
       case DOUBLE:
-        return dsr == Types.StringType.get() || dsr.typeId() == Type.TypeID.DECIMAL;
+        return dsr == Types.StringType.get() || dsr.typeId() == Type.TypeID.DECIMAL || dsr.typeId() == Type.TypeID.DECIMAL_FIXED;
       case DATE:
       case BINARY:
         return dsr == Types.StringType.get();
+      case DECIMAL_BYTES:
+        return isDecimalBytesUpdateAllowInternal(src, dsr);
       case DECIMAL:
-        if (dsr.typeId() == Type.TypeID.DECIMAL) {
-          Types.DecimalType decimalSrc = (Types.DecimalType)src;
-          Types.DecimalType decimalDsr = (Types.DecimalType)dsr;
-          if (decimalDsr.isWiderThan(decimalSrc)) {
-            return true;
-          }
-          if (decimalDsr.precision() >= decimalSrc.precision() && decimalDsr.scale() == decimalSrc.scale()) {
-            return true;
-          }
-        } else if (dsr.typeId() == Type.TypeID.STRING) {
-          return true;
-        }
-        break;
+      case DECIMAL_FIXED:
+        return isDecimalFixedUpdateAllowInternal(src, dsr);
```

Comment:
Is there internal schema evolution test coverage on `DECIMAL_BYTES` and `DECIMAL_FIXED`?

Reply:
There is some. I remember updating this because of test failures